File: system-test/java/src/main/java/com/optivem/eshop/dsl/driver/port/external/erp/ErpDriver.java
package com.optivem.eshop.dsl.driver.port.external.erp;

import com.optivem.eshop.dsl.driver.port.external.erp.dtos.GetProductRequest;
import com.optivem.eshop.dsl.driver.port.external.erp.dtos.GetProductResponse;
import com.optivem.eshop.dsl.driver.port.external.erp.dtos.ReturnsProductRequest;
import com.optivem.eshop.dsl.driver.port.shared.dtos.ErrorResponse;
import com.optivem.eshop.dsl.common.Result;

public interface ErpDriver extends AutoCloseable {
    Result<Void, ErrorResponse> goToErp();

    Result<GetProductResponse, ErrorResponse> getProduct(GetProductRequest request);

    Result<Void, ErrorResponse> returnsProduct(ReturnsProductRequest request);
}

File: system-test/typescript/src/testkit/driver/port/external/erp/erp-driver.ts
import type { Result } from '../../../../common/result.js';
import type { ErpErrorResponse } from './dtos/errors/ErpErrorResponse.js';
import type { GetProductRequest } from './dtos/GetProductRequest.js';
import type { GetProductResponse } from './dtos/GetProductResponse.js';
import type { ReturnsProductRequest } from './dtos/ReturnsProductRequest.js';
import type { ReturnsPromotionRequest } from './dtos/ReturnsPromotionRequest.js';

export interface ErpDriver {
  goToErp(): Promise<Result<void, ErpErrorResponse>>;
  getProduct(request: GetProductRequest): Promise<Result<GetProductResponse, ErpErrorResponse>>;
  returnsProduct(request: ReturnsProductRequest): Promise<Result<void, ErpErrorResponse>>;
  returnsPromotion(request: ReturnsPromotionRequest): Promise<Result<void, ErpErrorResponse>>;
  close(): Promise<void>;
}
